build/builder.go

```diff
@@ -998,5 +1005,14 @@ func (e deltaBranchSetError) Error() string {
 	return fmt.Sprintf("repository metadata in shard %q contains a different set of branch names than what was requested, which is unsupported in a delta shard build. old: %+v, new: %+v", e.shardName, e.old, e.new)
 }
 
+type deltaIndexOptionsMismatchError struct {
+	shardName  string
+	newOptions Options
+}
+
+func (e deltaIndexOptionsMismatchError) Error() string {
+	return fmt.Sprintf("One or more Index Options for shard %q does not match Builder's Index options; These index option updates are incompatible with delta build. new index options: %+v", e.shardName, e.newOptions)
```

Comment:
I'm printing all of the options for the current build. Alternatively I could just print the "critical" options that on update are incompatible with delta builds, or just print nothing.

Reply:
> I think changedOrRemovedFiles

I think we should only print the options which affect hashing.

Reply:
After thinking it over I also am leaning towards just printing the hash impacting options, since the error message is related to normal build vs delta build. I'm open to more feedback though.

I'm going to add a new method so that logging of these hashing options will reference the same Options as HashOptions(), so that changes to HashOptions will be consistent with logging in case we modify the selection of options for the hash.

Reply:
A follow up to my comment from a few days ago re: print all options vs. print just the Hash options: I've added a new type `HashOptions` so that we can easily print the struct for the two new log messages, and it will only print options that impact the Hash value, and not all options in `Options`.

Let me know what you think, there are other approaches that I'm open to. @stefanhengl @keegancsmith

Reply:
@stefanhengl

> I guess we only have the hash of the old index options and not the acutal values? Otherwise it would be great to just print the delta.
> 
> If we cannot have the delta, then let's print all options. It shouldn't happen too often and it helps us debugging. I think `changedOrRemovedFiles`, which might be very noisy to print, is only set later?

You're correct that we don't store the options used for the old hash (that is, the hash included as repo metadata that we store on the shard). I mentioned this @ggilmore and I can write a separate ticket for considering storing the options themselves as part of repo metadata. For this ticket I think keeping the hash value for identifying the mismatch in options will suffice and help keep scope reduced.

I didn't look into how noisy or high volume content the field `changedOrRemovedFiles` may get since I am leaning towardsslogging only the reduced options set; @ggilmore let us know what you think since that option is related to tombstones for delta builds